### PR TITLE
libsql-sys: bring back unix-excl-vfs feature

### LINKED ISSUE #833: Bring back `unix-excl-vfs` feature
We used to have a `unix-excl-vfs` feature, which got purged as part of 320855c1f4d461421c310f0c4927a3465da2fd89. We actually need to still maintain the feature, for use cases like @unikraft, where shared memory or file locks are not necessarily supported. The feature should just enforce that all libSQL connections use the `unix-excl` VFS, which does not rely on either shared memory mapping or file locks.

### PR DESCRIPTION
**Pull request by psarna**
It forces the unix-excl VFS to be used for connections, which makes the library work in environments without shared memory or file locks.

Fixes #833

### CODE REVIEW

**Review by penberg — APPROVED** [review]
Fine by me. @MarinPostma, comments?

File: libsql-server/src/replication/primary/logger.rs [hunk 1]
```diff
@@ -552,7 +552,16 @@ impl ReplicationLogger {
 
         let (new_frame_notifier, _) = watch::channel(generation_start_frame_no);
         unsafe {
-            let conn = rusqlite::Connection::open(db_path.join("data"))?;
```
Comment (MarinPostma) [inline libsql-server/src/replication/primary/logger.rs:555]: I realize this is a bit scary, why doesn't this conn taking a wal? Sounds like a bug!

Comment (psarna) [reply libsql-server/src/replication/primary/logger.rs:555]: hm, probably a preexisting one indeed. I'll add it

Comment (psarna) [reply libsql-server/src/replication/primary/logger.rs:555]: ah, it is preexisting, there's even a comment stating it's a bug. I think it's also not super trivial and we need to figure out the init order here -- this code checkpoints the db before bottomless init, and we need bottomless initialized for the wal manager to work if I remember correctly. I'll go ahead and merge the vfs fix for now to unblock the Unikraft folks, and create a separate issue for the preexisting bug

**Comment by psarna** [comment]
(just pushed a revert of the latest commit to try and pinpoint the test regression, because test failures only happen in github actions... I'll drop it soon)

**Comment by psarna** [comment]
ah, found one lack of `\0` sentinel in a str passed to C. maybe it's the reason!

**Comment by psarna** [comment]
hm, although it should be guarded by a feature.... so it's a bug, but probably not the direct regression cause

**Pull request merged**